code/game/objects/items/weapons/twohanded.dm

```diff
@@ -190,6 +190,22 @@
 		var/obj/structure/headpole/H = new (get_turf(src), W, src)
 		user.drop_item(W, H, force_drop = 1)
 
+/obj/item/weapon/spear/attack(var/mob/living/M, var/mob/user)
+	var/obj/item/I
+	if(user.zone_sel.selecting == "l_hand")
+		I = M.get_held_item_by_index(GRASP_LEFT_HAND)
+	else if(user.zone_sel.selecting == "r_hand")
+		I = M.get_held_item_by_index(GRASP_RIGHT_HAND)
+	if(I && istype(I,src.type) && user.a_intent == I_HELP)
+		if (prob(50))
+			playsound(user, 'sound/items/trayhit1.ogg', 50, 1)
+		else
+			playsound(user, 'sound/items/trayhit2.ogg', 50, 1)
```

Comment:
you could compact this into playsound(user, pick('sound/items/trayhit1.ogg', 'sound/items/trayhit2.ogg'), 50, 1

Reply:
Fixed.